packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -250,7 +250,8 @@ class FlutterCommandRunner extends CommandRunner<void> {
 
         globals.flutterVersion.ensureVersionFile();
         final bool machineFlag = topLevelResults['machine'] as bool;
-        if (topLevelResults.command?.name != 'upgrade' && topLevelResults['version-check'] as bool && !machineFlag) {
+        final bool ci = globals.platform.environment['CI'] == 'true';
```

Comment:
We already have an entire `BotDetector` class that does this:
https://github.com/flutter/flutter/blob/74bd7b6f6db6433783a36f94d9e43173745187f4/packages/flutter_tools/lib/src/base/bot_detector.dart#L53

Reply:
I was unaware. Of course there is!
Done